src/main/java/by/georgprog/epicmusicstore/service/album/AlbumServiceImpl.java

```diff
@@ -0,0 +1,101 @@
+package by.georgprog.epicmusicstore.service.album;
+
+import by.georgprog.epicmusicstore.config.security.UserDetailsImpl;
+import by.georgprog.epicmusicstore.dto.AlbumDto;
+import by.georgprog.epicmusicstore.exeption.forbidden.ObtainingDataException;
+import by.georgprog.epicmusicstore.exeption.unauthorized.AlbumNotFoundException;
+import by.georgprog.epicmusicstore.exeption.unauthorized.UserNotFoundException;
+import by.georgprog.epicmusicstore.mapper.AlbumMapper;
+import by.georgprog.epicmusicstore.model.AlbumEntity;
+import by.georgprog.epicmusicstore.repo.AlbumRepository;
+import by.georgprog.epicmusicstore.service.user.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class AlbumServiceImpl implements AlbumService {
+
+    private final UserService userService;
+    private final AlbumRepository albumRepository;
+    private final AlbumMapper albumMapper;
+
+    @Override
+    public List<AlbumEntity> findAll() {
+        return albumRepository.findAll();
+    }
+
+    @Override
+    public Optional<AlbumEntity> findById(Long id) {
+        return albumRepository.findById(id);
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        albumRepository.deleteById(id);
+    }
+
+    @Override
+    public void save(AlbumEntity entity) {
+        albumRepository.save(entity);
+    }
+
+    @Override
+    public AlbumDto findOwnerAlbum(Long id) throws AlbumNotFoundException, ObtainingDataException {
+        if (findById(id).isEmpty()) {
```

Comment:
Extract findById(id) result into a variable. We have to make as few requests to DB as possible